### PR TITLE
fix: nested maps on torch.compile for torch tensors

### PR DESCRIPTION
**Pull request by Killua7362**
torch.compile does not work well with dictionary of types. So changed types to str(types).

### CODE REVIEW

File: ivy/functional/ivy/nest.py [hunk 1]
```diff
@@ -1160,12 +1160,14 @@ def nested_map(
     to_ignore = ivy.default(to_ignore, ())
     extra_nest_types = ivy.default(extra_nest_types, ())
     if include_derived is True:
-        include_derived = {tuple: True, list: True, dict: True}
+        include_derived = {str(tuple): True, str(list): True, str(dict): True}
```
Comment (vedpatwardhan) [inline ivy/functional/ivy/nest.py:1163]: we could replace `str(tuple)` by `"tuple"` here and so on.

**Review by vedpatwardhan — CHANGES_REQUESTED** [review]
Hey @Killua7362, thanks for diving into this! I think if this fixes the issue, we can just refactor the `include_derived` argument to simply be a `Dict[str, bool]` at all places. So I'd suggest refactoring this function further along with all it's usages in the framework to instead pass a `"tuple"` instead of a `tuple`, `"list"` instead of `list` and so on 😄

File: ivy/functional/ivy/nest.py (hunk 1 above)
Comment (Killua7362) [reply ivy/functional/ivy/nest.py:1163]: I'll start working on it

**Review by vedpatwardhan — APPROVED** [review]
lgtm! Let's wait for the CI to complete running tests as it's quite a fundamental function, thanks @Killua7362 😄

**Comment by Killua7362** [comment]
> lgtm! Let's wait for the CI to complete running tests as it's quite a fundamental function, thanks @Killua7362 😄

Thank you Ved, let's hope nothing breaks XD

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Pull request merged**